src/main/java/conditions/CompareCondition.java

```diff
@@ -70,17 +70,28 @@ public Boolean apply( Map<String, String> row ) {
         // return <transformed leftColumn> <comparison> <transformed rightColumn>
 
         Object leftTransformed = row.get( leftColumn );
+        Object rightTransformed = row.get( rightColumn );
+
+        if ( !(leftColumn == null) && !( row.containsKey( leftColumn ) ) ) {
```

Comment:
cleaner to write `leftColumn != null` instead of `!(leftColumn == null)` (and same for below)